notebooks/hfdemo/tspulse/anomaly_detection/utility/datasets.py

```diff
@@ -0,0 +1,242 @@
+# Copyright contributors to the TSFM project
+#
+
+import numpy as np
+import torch
+import torch.utils.data
+
+
+epsilon = 1e-8
+
+
+class TSPulseReconstructionDataset(torch.utils.data.Dataset):
```

Comment:
Do we have a reconstruction dataset in the library itself? If this is general purpose, it should follow the same conventions as the rest of the datasets and be integrated within the library.